eisenhower-matrix/src/main/webapp/script.js

```diff
@@ -158,29 +123,50 @@ async function createCal() {
     var SCOPES = "https://www.googleapis.com/auth/calendar";
     fetch("https://www.googleapis.com/discovery/v1/apis/calendar/v3/rest") // Call the fetch function passing the url of the API as a parameter
     .then(function() {
+        //initializes api client
         gapi.load("client:auth2", initClient);
 
         function initClient() {
-            console.log("fetched google calendar api");
 
             gapi.client.init({
                 apiKey: API_KEY,
                 clientId: CLIENT_ID,
                 discoveryDocs: DISCOVERY_DOCS,
                 scope: SCOPES
             }).then(function () {
-                // Listen for sign-in state changes.
-                console.log(gapi.auth2.getAuthInstance().isSignedIn.get());
+                // Creates and adds new calendar that will contain scheduled tasks
                     var newCal = gapi.client.calendar.calendars.insert({
                         "summary": "Task Schedule",
                         "description": "Your tasks, scheduled by the Eisenhower Matrix."
                     }).then(function(response) {
+                        gapi.client.calendar.settings.get({setting:"timezone"}).then(
+                            function(response2) {
+                                //Timezone that calendar is operating in
+                                var calTimeZone = JSON.parse(response2.body).value;
+
+                                //Creates events for each task and adds them to calendar
+                                for (var t in Object.keys(tasks)){
+                                    gapi.client.calendar.events.insert({
```

Comment:
We also should check how to make this generic because almost all the API calls use a similar pattern. Did you have some permission errors?

Reply:
I agree, I'm not sure how to make it generic but whenever I tried anything else that may have been cleaner it gave me some issues. I haven't ran into any permission issues from my end though